app/helpers/application_helper.rb

```diff
@@ -205,8 +205,8 @@ def searchable?
   end
 
   def auto_complete_search(name, val, options = {})
-    path = options[:path] || send("#{controller_name}_path")
-    options.merge!(:class => "autocomplete-input form-control", :'data-url' => "#{path}/auto_complete_#{name}" )
+    path = options[:path] || send("#{controller_name}_path") + "/auto_complete_#{name}"
```

Comment:
I'd try and keep all three working, if that's not too much hassle.  Something like:

    path = options[:full_path]
    path ||= (options[:path] || send("#{controller_name}_path")) + "/auto_complete_#{name}"
    .... 'data-url' => path)

And then in filters/_form, pass :full_path in instead of :path.